### ISSUE TITLE
My project is broken because previous releases is removed in vpm repository

### ISSUE THREAD

**Issue by anatawa12** [VRCFury/VRCFury#13]
Please do not delete previous releases from vpm repository.

Removing previous releases from vpm repository breaks projects because vpm-resolver search for exactly same version as previously installed version as descrived in lock field in vpm-manifest.json.
This is very common behavior for package manager so this must not be a bug of vcc/vpm-resolver and I think problem in vrcfury side.
I think it's important for package manager (and repository) to make it possible to restore project exactly same as that time.

**Comment by SenkyDragon** [comment]
What's the failure mode? The reason we don't include previous versions in the repository is because downgrading vrcfury can be catastrophic. Corruption of components is basically guaranteed, because all unknown fields will be thrown out by unity during load, and then totally lost when the components are saved back out.

**Comment by anatawa12** [comment]
> What's the failure mode?

The error is simple, VPM says VRCFury 1.128.0 is not found.
Please try clone & open [OldVrcFuryTest](https://github.com/anatawa12/OldVrcFuryTest) or your avatars projects from remote git server.

```
[ERR] Could not get match for com.vrcfury.vrcfury 1.128.0
UnityEngine.Debug:LogError (object)
Serilog.Sinks.Unity3D.Unity3DLogEventSink:Emit (Serilog.Events.LogEvent)
Serilog.Core.Sinks.SafeAggregateSink:Emit (Serilog.Events.LogEvent)
Serilog.Core.Logger:Dispatch (Serilog.Events.LogEvent)
Serilog.Core.Logger:Write (Serilog.Events.LogEventLevel,System.Exception,string,object[])
Serilog.Core.Logger:Write (Serilog.Events.LogEventLevel,string,object[])
Serilog.Core.Logger:Error (string)
VRC.PackageManagement.VRCLibLogger:LogError (string)
VRC.PackageManagement.Core.Repos:GetPackageWithVersionMatch (string,string,System.Collections.Generic.IEnumerable`1<VRC.PackageManagement.Core.Types.Providers.IVRCPackageProvider>)
VRC.PackageManagement.Core.Repos:GetPackageWithVersionMatch (string,string)
VRC.PackageManagement.Core.Types.UnityProject:AddVPMPackage (string,string)
VRC.PackageManagement.Core.Types.Packages.VPMProjectManifest:Resolve (string)
VRC.PackageManagement.Resolver.Resolver:ResolveStatic (string) (at Packages/com.vrchat.core.vpm-resolver/Editor/Resolver/Resolver.cs:128)
VRC.PackageManagement.Resolver.Resolver/<CheckResolveNeeded>d__6:MoveNext () (at Packages/com.vrchat.core.vpm-resolver/Editor/Resolver/Resolver.cs:94)
UnityEngine.UnitySynchronizationContext:ExecuteTasks () (at /Users/bokken/buildslave/unity/build/Runtime/Export/Scripting/UnitySynchronizationContext.cs:107)
```


> The reason we don't include previous versions in the repository is because downgrading vrcfury can be catastrophic.

What I want to do is not downgrading. What I want to do is just restore installed version of vrcfury when I opened a project depends on a little old vrcfury got from git.

By design of VPM, contents of all vpm packages except for vpm-resolver will not be included in git repository. `.gitignore` file to do this will be included in template projects and automatically generated by vpm if missing.
Instead of contents of packages, the configuration (`vpm-manifest.json`) that holds exact version of all vpm packages is included and vpm-resolver will downloads exact version of the packages from vpm repository if they are missing.

To work this correctly, VPM repository must have all package versions. However, vrcfury's vpm repository doesn't have old packages so this does not work.

**Comment by SenkyDragon** [comment]
This issue has been fixed :). Apologies that previous versions are not available, but all versions are available after 1.137
